src/hotspot/os/linux/globals_linux.hpp

```diff
@@ -89,6 +89,13 @@
           "to disable both the override and the printouts."             \
           "See prctl(PR_SET_TIMERSLACK) for more info.")                \
                                                                         \
+  product(bool, PreventTHPsForThreadStacks, true, EXPERIMENTAL,         \
+          "If true, the JVM will attempt to prevent formation of "      \
+          "transparent huge pages in thread stacks.")                   \
```

Comment:
Can we expand this to state that it can be turned off by ergonomics. Or better yet perhaps this should be off by default and potentially turned on by ergonomics?

Reply:
As in, if THP=always, this will be turned on unless explicitly turned off? I dislike that I cannot see the default value in globals.hpp (for numerical args, "0" is always an indication for "no value", but for bool, it looks like the real default value).

What do you think about "AllowtTHPInThreadStacks" - default off, can be switched on manually to disable the mitigation if someone really wants THP for his thread stacks?

Alternative name "DisableTHPThreadStackMitigation" or somesuch, but I don't like the negation in the name.

Reply:
The value in globals.hpp cannot tell you what the "default" is because ergonomics may change it no matter which way we decide to express this. For a user that knows nothing about THP mode (e.g. me!) I can't tell whether this will be on or off or whether I may need to turn it off or on.

If the intent is that we normally enable the mitigation when THP mode = always, but the user can turn it off then something like `DisableTHPThreadStackMitigation` seems more appropriate to me.